src/main/java/de/uhd/ifi/se/decision/management/jira/extraction/versioncontrol/GitDiffedCodeExtractionManager.java

```diff
@@ -41,308 +40,298 @@
-					    + GitDecXtract.RAT_KEY_COMPONENTS_SEPARATOR + d.getKey();
-
-				    n.setKey(newKey);
-				    return n;
-				}).collect(Collectors.toList()));
-			    } else {
-				resultValues.addAll(editListEntry.getValue().stream().map(d -> {
-				    KnowledgeElement n = new KnowledgeElement();
-				    n.setDocumentationLocation(d.getDocumentationLocation());
-				    n.setDescription(d.getDescription());
-				    n.setId(d.getId());
-				    n.setSummary(d.getSummary());
-				    n.setType(d.getType());
-
-				    String newKey = newPath + GitDecXtract.RAT_KEY_COMPONENTS_SEPARATOR
-					    + dEntry.getValue().sequence + GitDecXtract.RAT_KEY_COMPONENTS_SEPARATOR
-					    + GitDecXtract.RAT_KEY_NOEDIT + GitDecXtract.RAT_KEY_COMPONENTS_SEPARATOR
-					    + d.getKey();
-
-				    n.setKey(newKey);
-				    return n;
-				}).collect(Collectors.toList()));
-			    }
+	private Diff diff;
+	private static final Logger LOGGER = LoggerFactory.getLogger(GitDiffedCodeExtractionManager.class);
+
+	// may include null values for keys!
+	// TODO Consider changing to Map<ChangedFile, CodeExtractionResult> or add
+	// CodeExtractionResult as an attribute to the ChangedFile class
+	private Map<DiffEntry, CodeExtractionResult> results = new HashMap<>();
+
+	public GitDiffedCodeExtractionManager(Diff diff) {
+		this.diff = diff;
+		processEntries();
+	}
+
+	public List<KnowledgeElement> getNewDecisionKnowledgeElements() {
+		return getNewOrOldDecisionKnowledgeElements(true);
+	}
+
+	public List<KnowledgeElement> getOldDecisionKnowledgeElements() {
+		return getNewOrOldDecisionKnowledgeElements(false);
+	}
+
+	private List<KnowledgeElement> getNewOrOldDecisionKnowledgeElements(boolean getNew) {
+		List<KnowledgeElement> resultValues = new ArrayList<>();
+		/*
+		 * @issue: one rationale modified by more than one edit line. Problem was found
+		 * in refs/remotes/origin/CONDEC-534.branch.filtering.improvements.RC2 An old
+		 * rationale located at(108:112:13) in file
+		 * ..extraction/versioncontrol/GitRepositoryFSManager.java was linked with two
+		 * change entries REPLACE(106-108,106-108) and REPLACE(109-114,109-120)
+		 * 
+		 * Such rationale would be touched twice in below streams, making its key
+		 * unusable.
+		 * 
+		 * @alternative: streams will return reference rationale, but should somehow try
+		 * not to modify the rationale key more than once!
+		 * 
+		 * @con: it would not be possible to see that many edits changed rationale.
+		 * 
+		 * @alternative: streams will return new origin objects in case of detected
+		 * repetition new object will be created!
+		 * 
+		 * @con: more code needs to be changed.
+		 * 
+		 * @pro: information about change by more than one edit will be preserved.
+		 * 
+		 * @decision: streams will return new rationale objects and never use origin
+		 * references!
+		 * 
+		 * @pro: information about change by more than one edit will be preserved.
+		 * 
+		 */
+		if (results.size() > 0) {
```

Comment:
Die Cyclomatische-Komplexität ist hier sehr hoch vlt. kann man den teil in den for-Schleifen besser rausrefaktorisieren